### PR TITLE
NOISSUE - Remove vendor dir

### LINKED ISSUE #2205: Feature: coap-cli: Remove vendor dir
### Is your feature request related to a problem? Please describe.

Remove the vendor directory

### Describe the feature you are requesting, as well as the possible use case(s) for it.

To remove the vendor directory for more consistent vendoring

### Indicate the importance of this feature to you.

Should-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a dependency update as it removes the vendor directory.

## What does this do?
Removes vendor directory and modifys build mode to match this change.

## Which issue(s) does this PR fix/relate to?
Resolves https://github.com/absmach/magistrala/issues/2205

## Have you included tests for your changes?
No

## Did you document any new/modified feature?
N/A

### Notes
None

Signed-off-by:  [1998-felix](mailto:felix.gateru@gmail.com)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 all:
-	CGO_ENABLED=0 GOOS=linux go build -mod=vendor -ldflags "-s -w" -o build/coap-cli-linux cmd/main.go
-	CGO_ENABLED=0 GOOS=darwin go build -mod=vendor -ldflags "-s -w" -o build/coap-cli-darwin cmd/main.go
-	CGO_ENABLED=0 GOOS=windows go build -mod=vendor -ldflags "-s -w" -o build/coap-cli-windows cmd/main.go
\ No newline at end of file
+	CGO_ENABLED=0 GOOS=linux go build -ldflags "-s -w" -o build/coap-cli-linux cmd/main.go
+	CGO_ENABLED=0 GOOS=darwin go build -ldflags "-s -w" -o build/coap-cli-darwin cmd/main.go
+	CGO_ENABLED=0 GOOS=windows go build -ldflags "-s -w" -o build/coap-cli-windows cmd/main.go
```
Comment (rodneyosodo) [inline Makefile]: Add a new line

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,12 +1,20 @@
 module github.com/mainflux/coap-cli
```
Comment (rodneyosodo) [inline go.mod]: Fix module name

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: go.mod [hunk 3]
```diff
@@ -1,12 +1,18 @@
-module github.com/mainflux/coap-cli
+module github.com/absmach/coap-cli
 
-go 1.15
+go 1.21
```
Comment (rodneyosodo) [inline go.mod]:
```suggestion
go 1.22
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**